### PR TITLE
Handle Wikimedia server errors nicely

### PR DESCRIPTION
**Pull request by MaxSem**
Bug: https://phabricator.wikimedia.org/T222932

### CODE REVIEW

**Comment by samwilson** [comment]
With regard to using HtmlFormatter, do you want to comment on #153? Seems that maybe it will do everything we need for parsing MediaWiki HTML elsewhere in this tool as well, so maybe that can be resolved.

**Comment by MaxSem** [comment]
I'm using HtmlFormatter only to read error HTML into DOM, the question in that ticket is much more global.

File: utils/utils.php [hunk 1]
```diff
@@ -197,3 +200,43 @@ function formatException( Exception $ex ): string {
...
+	}
+
+	// Otherwise, try hhvm-fatal-error.php style
+	if ( !$text ) {
+		foreach ( $doc->getElementsByTagName( 'code' ) as $node ) {
+			$text = trim( $text . "\n" . $node->textContent );
+		}
+	}
+
+	return $text ? "Wikivoyage servers returned an error: $text" : null;
```
Comment (samwilson) [inline utils/utils.php]: Wikivoyage? :)

File: utils/Api.php [hunk 2]
```diff
@@ -76,10 +75,10 @@ function ( ResponseInterface $response ) {
 					throw new HttpException( 'HTTP error ' . $response->getStatusCode(), $response->getStatusCode() );
 				}
 				return $response->getBody()->getContents();
-			},
+			}/*,
 			function ( RequestException $e ) {
```
[the remark this answers is not in the document]
Comment (MaxSem) [reply utils/Api.php]: Done.

File: utils/utils.php (hunk 1 above)
Comment (MaxSem) [reply utils/utils.php]: You don't like Wikivoyage, eh?

**Pull request merged**